subprojects/ce-login/cli/CliCeLoginV1.cpp

```diff
@@ -541,3 +552,54 @@ CeLogin::CeLoginRc CeLogin::decodeAndVerifyCeLoginHsfV1(
     }
     return sRc;
 }
+
+CeLogin::CeLoginRc
+    CeLogin::generateRandomPassword(char* dstParm,
+                                    const uint64_t requestPasswordSize)
+{
+    // Generate an array of the "usable" characters for ACF passwords. A random
+    // password will be generated by randomly selecting characters in this list.
+    // clang-format off
+    const char sValidCharacters[] = {
+        'A','B','C','D','E','F','G','H',/*'I',*/'J','K',  'L',  'M','N',/*'O',*/'P','Q','R','S','T','U','V','W','X','Y','Z',
+        'a','b','c','d','e','f','g','h',  'i',  'j','k',/*'l',*/'m','n',  'o',  'p','q','r','s','t','u','v','w','x','y','z',
+        '0','1','2','3','4','5','6','7','8','9'
+    };
+    // clang-format on
+
+    for (uint64_t sIdx = 0; sIdx < requestPasswordSize; sIdx++)
+    {
+        // Reuse the memory to store the temporary value
+        uint8_t& sRandomByte = ((uint8_t*)dstParm)[sIdx];
+        do
+        {
+            int rc = RAND_priv_bytes(&sRandomByte, sizeof(sRandomByte));
+            if (1 != rc)
+            {
+                return CeLoginRc::Failure;
+            }
+            // To simplify the code, reject a random number that is an
+            // out-of-bounds index.
+        } while (sRandomByte >= sizeof(sValidCharacters));
```

Comment:
Obligatory comment about the inefficiency of rejecting > 75% of the generated values. It's not wrong though; the resulting entropy is unaffected.

Reply:
Agreed. In my mind it was a balance between the code complexity of accepting multiples of `sizeof(sValidCharacters)` while rejecting any "remainder" values that would weigh lower indexed values more than later indexed.

In this case we are only creating 10 characters at ACF creation, so I think code readability/review-ability is more important than efficiency.